### PR TITLE
Enhance the messaging when connecting through SSO.

### PR DESCRIPTION
**Pull request by nervoustwit**
Using jquery, check the availability of a username that a user chooses when trying to connect as a user for the first time on /entry/connect.

Requires the following: 
 
 - [x] Create a meaningful javascript variable to be added to to gdn.meta to tell the script to hide the password field.
 - [x] Make messaging cleaner by extracting into function.
 - [x] Optionally pass message from gdn.meta for better translation.
 - [x] Stop form from submitting with used username.

### CODE REVIEW

**Comment by nervoustwit** [comment]
Make hiding password on connect a configurable option, add configurable error message on connect, do not allow form to be submitted by pressing enter.

File: applications/dashboard/js/entry.js [hunk 1]
```diff
@@ -87,8 +99,19 @@ jQuery(document).ready(function($) {
...
+        if($(".Messages.Errors").length) {
+            $(".Messages.Errors").html("<ul><li>" + msg + "</li></ul>");
+        } else {
+            $('#Form_ConnectName').closest('form').prepend("<div class='Messages Errors'><ul><li>" + msg + "</li></ul></div>");
+        }
+    }
+
     checkConnectName();
-    $('#Form_ConnectName').keyup(checkConnectName);
+    $('#Form_ConnectName').blur(checkConnectName);
```
Comment (linc) [inline applications/dashboard/js/entry.js]: https://github.com/vanilla/vanilla/pull/3522

**Comment by linc** [comment]
Why are we blocking the Enter key?

File: applications/dashboard/js/entry.js [hunk 2]
```diff
@@ -69,12 +77,16 @@ jQuery(document).ready(function($) {
...
                             if (fineprint.length) {
-                                fineprint.html(gdn.definition('Choose a name to identify yourself on the site.'));
+                                fineprint.html(gdn.getMeta('entryChooseNameMsg', 'Choose a name to identify yourself on the site.'));
                             }
                         } else {
-                            $('#ConnectPassword').show();
-                            if (fineprint.length) {
-                                fineprint.html(gdn.definition('Username already exists.'));
+                            // If the username is not available, and the client does not want users to take over existing accounts, generate an error message and empty the input field.
+                            if(gdn.definition('NoConnectName', true)) {// && gdn.definition('ForceCreateConnectName', false)
```
Comment (linc) [inline applications/dashboard/js/entry.js]: space.

File: applications/dashboard/js/entry.js [hunk 3]
```diff
@@ -1,6 +1,14 @@
 
 // This file contains javascript that is specific to the dashboard/entry controller.
 jQuery(document).ready(function($) {
+    // For the SSO connect user page, do not allow to submit by pressing enter.
+    $("#dashboard_entry_connect").keydown(function(event){
+        if(event.keyCode == 13) {
```
Comment (linc) [inline applications/dashboard/js/entry.js]: space.

**Comment by linc** [comment]
You are reversing too much of https://github.com/vanilla/vanilla/pull/3522

**Comment by nervoustwit** [comment]
Closing this, we are handling this entirely with a config setting that is read by the controller and the view. Leaving javascript as is. See: https://github.com/vanilla/vanilla/pull/3557

**Pull request closed without merging**